apps/namadillo/src/atoms/registry/functions.ts

```diff
@@ -0,0 +1,46 @@
+import { Chain } from "@chain-registry/types";
+import { workingRpcsAtom } from "atoms/integrations";
+import { getDefaultStore } from "jotai";
+
+export const getRandomRpcAddress = (chain: Chain): string => {
+  const availableRpcs = chain.apis?.rpc;
+  if (!availableRpcs) {
+    throw new Error("There are no available RPCs for " + chain.chain_name);
+  }
+
+  const randomRpc =
+    availableRpcs[Math.floor(Math.random() * availableRpcs.length)];
+
+  return randomRpc.address;
+};
+
+// eslint-disable-next-line
+
+type QueryFn<T> = (rpc: string) => Promise<T>;
+
+export const queryAndStoreRpc = async <T>(
```

Comment:
This is cool! I think you're right about using the RPCs sequentially instead of randomly. Just from brief testing, it seems hit or miss whether an RPC will actually work (and I'm hoping they are ordered roughly by reliability).